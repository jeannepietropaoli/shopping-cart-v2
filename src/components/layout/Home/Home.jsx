import "../../../styles/Home.css";
import ProductCarousel from "../../shared/ProductCarousel";
import { Link } from "react-router-dom";

export default function Home({products, error, loading}) {
  return (
    <section className="home">
      <div className="container">
        <h1 className="title">Pack you fruits.</h1>
        <p>Remember to eat your fruits by packing them for the day. Start collecting them in your cart and earn rewards when you go to check out and actually eat them !</p>
        <button><Link to='/shop'>Shop now</Link></button>
        <ProductCarousel products={products} error={error} loading={loading}/> 
      </div>
    </section>
  );
}
