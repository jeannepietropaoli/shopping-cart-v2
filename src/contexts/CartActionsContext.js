import { createContext } from 'react';

export const CartActionsContext = createContext();