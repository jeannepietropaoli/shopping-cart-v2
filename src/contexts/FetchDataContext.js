import { createContext } from "react";

export const FetchDataContext = createContext();