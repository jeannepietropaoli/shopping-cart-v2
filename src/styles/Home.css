.home {
    flex-grow: 1;
    display: flex;
    flex-direction: column;
}

.home__container {
    text-align: center;
    flex-grow: 1;
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    align-items: center;
    gap: 18px;
}

.home__title {
    font-family: Helvetica, sans-serif;
    font-size: 2.5rem;
    letter-spacing: -2px;
    margin-top: 8px;
}

.home__text-container {
    display: flex;
    flex-direction: column;
    gap: 10px;
}