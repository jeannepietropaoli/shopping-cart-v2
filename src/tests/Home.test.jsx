import { render, screen } from "@testing-library/react";
import Home from "../components/layout/Home/Home";
import { describe, expect, it, vi } from "vitest";
import { MemoryRouter } from 'react-router-dom';
import { ProductsContext } from "../contexts/ProductsContext";

function renderHomeComponent() {
    return render(
        <MemoryRouter>
            <ProductsContext.Provider value={{
                products: [{id:1, name: 'product1', image:'', price: 2},
                {id:2, name: 'product2', image:'', price: 8},
                {id:3, name: 'product3', image:'', price: 3}],
                error:false,
                loading:false
            }}>
                <Home />
            </ProductsContext.Provider>
        </MemoryRouter>
    )
}

describe('Testing Home Component', () => {
    it('renders the component correctly', () => {
        const { container } = renderHomeComponent() // The Link component from react-router has to be inside a router for it to work
        screen.debug()
        expect(container).toMatchSnapshot()
    })

    it('renders the h1', () => {
        renderHomeComponent()
        const h1 = screen.getByRole('heading', {level: 1})
        expect(h1).toBeInTheDocument()
        expect(h1.textContent).toMatch('Pack your fruits.')
    })

    it('renders a Shop now button linked to he /shop path', () => {
        renderHomeComponent()
        const shopLinkBtn = screen.getByRole('button', {name: 'Shop now'})
        expect(shopLinkBtn).toBeInTheDocument()
        const shopLink = screen.getByRole('link', {name: 'Shop now'})
        expect(shopLink).toBeInTheDocument()
        expect(shopLink).toHaveAttribute('href', '/shop')
    })

    it('renders a ProductCarousel component if no error', () => {
        renderHomeComponent()
        const carousel = screen.getByTestId('carousel')
        expect(carousel).toBeInTheDocument()
    })
})