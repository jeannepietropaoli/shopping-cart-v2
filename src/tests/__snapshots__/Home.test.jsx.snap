// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing Home Component > renders the component correctly 1`] = `
<div>
  <section
    class="home"
  >
    <div
      class="container home__container"
    >
      <h1
        class="home__title"
      >
        Pack your fruits.
      </h1>
      <div
        class="home__text-container"
      >
        <p>
          Remember to eat your fruits by packing them for the day. Start collecting them in your cart and earn rewards when you go to check out and actually eat them !
        </p>
        <p>
          Please note that prices follow the market fluctuations and can change when you hit refresh.
        </p>
      </div>
      <button>
        <a
          href="/shop"
        >
          Shop now
        </a>
      </button>
      <div
        class="carousel"
        data-testid="carousel"
      >
        <div
          class="carousel__slider"
        >
          <div
            class="carousel__item"
            data-testid="carousel-item"
          >
            <img
              alt="product1's emoji"
              class="carousel__img"
              src=""
            />
            <p
              class="carousel__infos"
            >
              product1
            </p>
          </div>
          <div
            class="carousel__item"
            data-testid="carousel-item"
          >
            <img
              alt="product2's emoji"
              class="carousel__img"
              src=""
            />
            <p
              class="carousel__infos"
            >
              product2
            </p>
          </div>
          <div
            class="carousel__item"
            data-testid="carousel-item"
          >
            <img
              alt="product3's emoji"
              class="carousel__img"
              src=""
            />
            <p
              class="carousel__infos"
            >
              product3
            </p>
          </div>
        </div>
        <div
          class="carousel__slider"
        >
          <div
            class="carousel__item"
            data-testid="carousel-item"
          >
            <img
              alt="product1's emoji"
              class="carousel__img"
              src=""
            />
            <p
              class="carousel__infos"
            >
              product1
            </p>
          </div>
          <div
            class="carousel__item"
            data-testid="carousel-item"
          >
            <img
              alt="product2's emoji"
              class="carousel__img"
              src=""
            />
            <p
              class="carousel__infos"
            >
              product2
            </p>
          </div>
          <div
            class="carousel__item"
            data-testid="carousel-item"
          >
            <img
              alt="product3's emoji"
              class="carousel__img"
              src=""
            />
            <p
              class="carousel__infos"
            >
              product3
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
